build.gradle

```
@@ -130,21 +143,28 @@ bintray {
     publish = true
 }
 
-generateGrammarSource {
-    arguments += ["-package", "com.exactpro.th2.proto.service.generator.core.antlr"]
+compileKotlin {
+    kotlinOptions.jvmTarget = "11"
 }
 
-compileJava.dependsOn.add('generateGrammarSource')
-
-gradlePlugin {
-    plugins {
-        serviceGeneratorPlugin {
-            id = 'com.exactpro.th2.service-generator'
-            implementationClass = 'com.exactpro.th2.proto.service.generator.plugin.ServiceGeneratorPlugin'
-        }
+mainClassName = "com.exactpro.th2.service.generator.protoc.Main"
+
+shadowJar {
+    //mergeServiceFiles()
+
+    manifest {
+        attributes(
+                'Created-By':               "${System.getProperty('java.version')} (${System.getProperty('java.vendor')})",
+                'Specification-Title':      '',
+                'Specification-Vendor':     'Exactpro Systems LLC',
+                'Implementation-Title':     project.archivesBaseName,
+                'Implementation-Vendor':    'Exactpro Systems LLC',
+                'Implementation-Vendor-Id': 'com.exactpro',
+                'Implementation-Version':   project.version
+        )
     }
+    
+    archiveName = project.name
+    destinationDir = "$buildDir/artifacts" as File
```

Those properties are depricated. Please, use these ones
```suggestion
    archiveFileName = project.name
    destinationDirectory = "$buildDir/artifacts" as File
```